### DISCUSSION TITLE
should core templates work this way?

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#4937] [Q&A Core] [answered]
I was playing with the admin template and I noticed a very weird thing.   Could somebody look at it and confirm it?  
I don't ask for any change just confirmation that this is the correct behavior.

I know that cache needs to be cleared after any change in code. I did it.

Admin template header is used this way:

`$ADMIN_HEADER = e107::getCoreTemplate('admin', 'header', false);`

**1Q. help to getCoreTemplate() is outdated?** 

this line:   e107_themes/templates/user_template.php (if override not found or $override is false)  
```
 	 * Example usage: <code>e107::getCoreTemplate('user', 'short_start');</code>
	 * Will search for:
	 * - e107_themes/current_frontend_theme/templates/user_template.php (if $override is true)
	 * - e107_themes/templates/user_template.php (if override not found or $override is false)
	 * - $USER_TEMPLATE array which contains all user templates
	 * - $USER_TEMPLATE['short_start'] (if key is null, $USER_TEMPLATE will be returned)
```

**2Q $override parameter for all keys has to be the same?** 

I tried to change this parameter for the "header" key: 
 
`$ADMIN_HEADER = e107::getCoreTemplate('admin', 'header', true);`

no change, $reg_path is still core/e107/templates/admin   and $override is still false inside e107::getCoreTemplate() method

I found that the first calling is with key "nav"  in admin_shortcodes, after changing $override there, it was changed for the header too.

$reg_path  is correct then:  core/e107/templates/admin/ext  

Maybe this is clearer:
I copied admin_template.php to bootstrap3/templates folder and changed something to know that the theme version is loaded. 

Combination:
```
$template = e107::getCoreTemplate('admin', 'nav', false);
$ADMIN_HEADER = e107::getCoreTemplate('admin', 'header', true);
```
all $overrides inside getCoreTemplate() ale false, path is core/e107/templates/admin for header too

Combination:
```
$template = e107::getCoreTemplate('admin', 'nav', true);
$ADMIN_HEADER = e107::getCoreTemplate('admin', 'header', false);
```
correct, core/e107/templates/admin/ext + core/e107/templates/admin/ for header

Combination:
```
$template = e107::getCoreTemplate('admin', 'nav', true);
$ADMIN_HEADER = e107::getCoreTemplate('admin', 'header', true);
```
correct, core/e107/templates/admin/ext for both keys

So should it work this way?  Or this is just a way how to forbid overriding core templates?  

Thanks

**Comment by Moc** [comment, marked as answer]
Core templates should work this way, however there is currently no support for customizing the Admin Area. So there is hardcoded values used to prevent customization.
